core/src/main/scala/com/cloudera/spark/CGroupsMemoryHandle.scala

```diff
@@ -0,0 +1,86 @@
+package com.cloudera.spark
+
+import com.cloudera.spark.CGroupsMemoryHandle.{cGroupMemoryStatPath, getCGroupMemStats}
+
+import scala.io.Source
+
+class CGroupsMemoryHandle(cgroupMemStats: Map[String, Long]) extends MemoryGetter {
+
+  var orderedNames: Option[Seq[String]] = None
+
+  override val namesAndReporting: Seq[(String, PeakReporting)] = {
+    val namesAndReporting = cgroupMemStats.toSeq.map { x =>
+      if (Set("pgpgin", "pgpgout", "total_pgpgin", "total_pgpgout")(x._1)) {
+        (x._1, IncrementCounts)
+      } else {
+        (x._1, IncrementBytes)
+      }
+    }
+    orderedNames = Some(namesAndReporting.map { x => x._1})
```

Comment:
so I don't want to spend too much time on style of this code, as its not terribly important, but I think this could be cleaner so just want to make some suggestions for you to consider.

If you're going to define a hard-coded set of metrics, then I'd also reuse that for the `namesAndOrdering`.  Then you also avoid the `var` for `orderedNames`, which just makes it a little harder to track what is going on.  And the iterating over the values when assigning them becomes easier too.

```scala
object CGroupsMemoryHandle {
  
  val metrics = Seq( "cache", ...)
    .flatMap { metric => Seq(metric, "total_" + metric) }
  
val countMetrics = Set("pgpgin", "pgpout", "total_pgpgin", "total_pgpout")
  ...
}

class CGroupMemoryHandle extends MemoryGetter {

  override val namesAndReporting = CGroupMemoryHandle.metrics.map { m =>
    val peakReporting = if (CGroupMemoryHandle.countMetrics(m)) IncrementCounts else IncrementBytes
    (m, peakReporting)
  }
  
  override def values(dest: Array[Long], offset: Int): Unit = {
    var idx = offset
    val metricsSnapshot = ... // get metrics here
    CGroupMemoryHandle.metrics.foreach { m =>
      metricsSnapshot.get(m).foreach { newValue =>
        dest(idx) = newValue
      }
      idx += 1
    }
  }
```

Reply:
Fixed in the latest commit.